lib/puppet/type/cisco_stp_global.rb

```diff
@@ -0,0 +1,753 @@
+    So for bd_designated_priority, the bridge domain range 2 to 42 will be set
+    to 40960 and 83to 92 and 1000 to 2300 will be set to 53248
+
+    Example:
+    cisco_stp_global { 'default':
+      bd_designated_priority       => [['2-42', '40960'], ['83-92,1000-2300', '53248']],
+      bd_forward_time              => [['2-42', '26'], ['83-92,1000-2300', '20']],
+      bd_hello_time                => [['2-42', '6'], ['83-92,1000-2300', '9']],
+      bd_max_age                   => [['2-42', '26'], ['83-92,1000-2300', '20']],
+      bd_priority                  => [['2-42', '40960'], ['83-92,1000-2300', '53248']],
+      bd_root_priority             => [['2-42', '40960'], ['83-92,1000-2300', '53248']],
+      bpdufilter                   => true,
+      bpduguard                    => true,
+      bridge_assurance             => false,
+      domain                       => 100,
+      fcoe                         => false,
+      loopguard                    => true,
+      mode                         => 'mst',
+      mst_designated_priority      => [['2-42', '40960'], ['83-92,1000-2300', '53248']],
+      mst_forward_time             => 25,
+      mst_hello_time               => 5,
+      mst_inst_vlan_map            => [['2', '6-47'], ['92', '120-400']],
+      mst_max_age                  => 35,
+      mst_max_hops                 => 200,
+      mst_name                     => 'nexus',
+      mst_priority                 => [['2-42', '40960'], ['83-92,1000-2300', '53248']],
+      mst_revision                 => 34,
+      mst_root_priority            => [['2-42', '40960'], ['83-92,1000-2300', '53248']],
+      pathcost                     => 'long',
+      vlan_designated_priority     => [['1-42', '40960'], ['83-92,1000-2300', '53248']],
+      vlan_forward_time            => [['1-42', '19'], ['83-92,1000-2300', '13']],
+      vlan_hello_time              => [['1-42', '10'], ['83-92,1000-2300', '6']],
+      vlan_max_age                 => [['1-42', '21'], ['83-92,1000-2300', '13']],
+      vlan_priority                => [['1-42', '40960'], ['83-92,1000-2300', '53248']],
+      vlan_root_priority           => [['1-42', '40960'], ['83-92,1000-2300', '53248']],
+    }
+  "
+  ###################
+  # Resource Naming #
+  ###################
+
+  newparam(:name, namevar: :true) do
+    desc 'ID of the stp global config. Valid values are default.'
+
+    validate do |inst_name|
+      fail "only acceptable name is 'default'" if inst_name != 'default'
+    end
+  end # param id
+
+  ##############
+  # Attributes #
+  ##############
+
+  # rubocop:disable Style/PredicateName
+  newproperty(:bd_designated_priority, array_matching: :all) do
+    format = '[[bd_inst_list, designated_priority], [bdil, dp]]'
+    desc 'An array of [bd_inst_list, designated_priority] pairs. '\
+         "Valid values match format #{format}."
+
+    # Override puppet's insync method, which checks whether current value is
```

Comment:
Discussed this with Mike also, I originally tried to stop repeating the code and called a common method but there was some name space issue (I think, it was a moth back or so) possibly because I am using 'should' inside the insync?(is), not very sure. But anyway, I did try this and it was not working. Can we put this in a different user story and move with this PR for now? If not, please provide some example code or if you can point me to a provider which does this kind of override, I can try again.